utils/ccloud_library.sh

```diff
@@ -708,29 +708,26 @@ function ccloud::set_kafka_cluster_use() {
 function ccloud::create_ccloud_stack() {
   enable_ksql=$1
 
-  RANDOM_NUM=$((1 + RANDOM % 1000000))
-  #echo "RANDOM_NUM: $RANDOM_NUM"
+  if [[ -z "$SERVICE_ACCOUNT_ID" ]]; then
+    # Service Account is not received so it will be created
+    local RANDOM_NUM=$((1 + RANDOM % 1000000))
+    SERVICE_NAME=${SERVICE_NAME:-"demo-app-$RANDOM_NUM"}
+    SERVICE_ACCOUNT_ID=$(ccloud::create_service_account $SERVICE_NAME)
+    
+  fi
```

Comment:
After this line, I believe that we should check for empty SERVICE_NAME and exit with a message if it's not provided or wasn't calculated in the case of the SERVICE_ACCOUNT_ID being provided.

Reply:
Added the logic in, thx!